### PR TITLE
add cicerone navigation

### PR DESCRIPTION
**Pull request by tosh17**

### CODE REVIEW

File: app/src/main/java/ru/thstdio/aa2020/navigation/Screen.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    fun DetailsCinema(idCinema: Long) =
+        FragmentScreen(key = ScreenKey.DetailsCinema.name) {
+            FragmentMoviesDetails.newInstance(
+                idCinema
+            )
+        }
+}
+
+enum class ScreenKey {
```
Comment (rHlushan) [inline app/src/main/java/ru/thstdio/aa2020/navigation/Screen.kt]: what for?

File: app/src/main/java/ru/thstdio/aa2020/navigation/Screen.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package ru.thstdio.aa2020.navigation
+
+import com.github.terrakok.cicerone.androidx.FragmentScreen
+import ru.thstdio.aa2020.ui.detail.FragmentMoviesDetails
+import ru.thstdio.aa2020.ui.list.FragmentMoviesList
+
+object Screen {
```
Comment (rHlushan) [inline app/src/main/java/ru/thstdio/aa2020/navigation/Screen.kt]: why u need this object?

File: app/src/main/java/ru/thstdio/aa2020/ui/CinemaApp.kt [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
+import com.github.terrakok.cicerone.Cicerone
+import com.github.terrakok.cicerone.NavigatorHolder
 import ru.thstdio.aa2020.api.Repository
 
 class CinemaApp : Application() {
+    private val cicerone = Cicerone.create()
+    val router get() = cicerone.router
+    val navigatorHolder: NavigatorHolder
+        get() = cicerone.getNavigatorHolder()
+    val repository: Repository = Repository()
```
Comment (rHlushan) [inline app/src/main/java/ru/thstdio/aa2020/ui/CinemaApp.kt]: maybe by lazy{}

File: app/src/main/java/ru/thstdio/aa2020/ui/CinemaApp.kt [hunk 4]
```diff
@@ -1,10 +1,24 @@
...
+
     companion object {
```
Comment (rHlushan) [inline app/src/main/java/ru/thstdio/aa2020/ui/CinemaApp.kt]: avoid this companion
and cast Context.applicationContext to CinemaApp

File: app/src/main/java/ru/thstdio/aa2020/ui/MainActivity.kt [hunk 5]
```diff
@@ -2,32 +2,32 @@ package ru.thstdio.aa2020.ui
...
+import com.github.terrakok.cicerone.Router
+import com.github.terrakok.cicerone.androidx.AppNavigator
 import ru.thstdio.aa2020.R
-import ru.thstdio.aa2020.ui.detail.FragmentMoviesDetails
-import ru.thstdio.aa2020.ui.list.FragmentMoviesList
+import ru.thstdio.aa2020.navigation.Screen
 
-class MainActivity : AppCompatActivity(), Navigation {
+class MainActivity : AppCompatActivity() {
+    private val navigator = AppNavigator(this, R.id.fragment_container_view)
```
Comment (rHlushan) [inline app/src/main/java/ru/thstdio/aa2020/ui/MainActivity.kt]: u don't need to keep navigator reference

File: app/src/main/java/ru/thstdio/aa2020/ui/detail/FragmentMoviesDetails.kt [hunk 6]
```diff
@@ -38,7 +39,7 @@ class FragmentMoviesDetails : FragmentNavigation(R.layout.fragment_movies_detail
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding.areaBack.setOnClickListener {
-            router.back()
+            router.backTo(Screen.ListCinema())
```
Comment (rHlushan) [inline app/src/main/java/ru/thstdio/aa2020/ui/detail/FragmentMoviesDetails.kt]: u don't need backTo here

File: app/src/main/java/ru/thstdio/aa2020/ui/CinemaApp.kt (hunk 3 above)
Comment (tosh17) [reply app/src/main/java/ru/thstdio/aa2020/ui/CinemaApp.kt]: fix

File: app/src/main/java/ru/thstdio/aa2020/ui/CinemaApp.kt (hunk 4 above)
Comment (rHlushan) [reply app/src/main/java/ru/thstdio/aa2020/ui/CinemaApp.kt]: ?

File: app/src/main/java/ru/thstdio/aa2020/ui/detail/MoviesDetailsScreen.kt [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package ru.thstdio.aa2020.ui.detail
+
+import com.github.terrakok.cicerone.androidx.FragmentScreen
+
+object MoviesDetailsScreen {
```
Comment (rHlushan) [inline app/src/main/java/ru/thstdio/aa2020/ui/detail/MoviesDetailsScreen.kt]: what sense of object here?
maybe just
class MoviesDetailsScreen : FragmentScreen

File: app/src/main/java/ru/thstdio/aa2020/ui/list/MoviesListScreen.kt [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package ru.thstdio.aa2020.ui.list
+
+import com.github.terrakok.cicerone.androidx.FragmentScreen
+
+object MoviesListScreen {
```
Comment (rHlushan) [inline app/src/main/java/ru/thstdio/aa2020/ui/list/MoviesListScreen.kt]: same

File: app/src/main/java/ru/thstdio/aa2020/ui/CinemaApp.kt (hunk 4 above)
Comment (rHlushan) [reply app/src/main/java/ru/thstdio/aa2020/ui/CinemaApp.kt]: ???

**Review by rHlushan — APPROVED** [review]

**Pull request merged**